base/cvd/cuttlefish/host/commands/cvd/cli/frontline_parser.cpp

```diff
@@ -17,119 +17,53 @@
 #include "host/commands/cvd/cli/frontline_parser.h"
 
 #include <memory>
+#include <optional>
+#include <string>
 #include <vector>
 
 #include <android-base/file.h>
 #include <android-base/strings.h>
 
 #include "host/commands/cvd/cli/flag.h"
+#include "host/commands/cvd/cli/selector/arguments_separator.h"
 #include "host/commands/cvd/cli/selector/selector_constants.h"
 #include "host/commands/cvd/cli/types.h"
 
 namespace cuttlefish {
 
-Result<cvd_common::Args> ExtractCvdArgs(cvd_common::Args& args) {
-  FrontlineParser::ParserParam server_param{
-      .server_supported_subcmds = {"*"},
-      .all_args = args
-  };
-  auto frontline_parser = CF_EXPECT(FrontlineParser::Parse(server_param));
-  CF_EXPECT(frontline_parser != nullptr);
-
-  const auto prog_path = frontline_parser->ProgPath();
-  const auto new_sub_cmd = frontline_parser->SubCmd();
-  cvd_common::Args cmd_args{frontline_parser->SubCmdArgs()};
-
-  cvd_common::Args new_exec_args{prog_path};
-  if (new_sub_cmd) {
-    new_exec_args.push_back(*new_sub_cmd);
-  }
-  new_exec_args.insert(new_exec_args.end(), cmd_args.begin(), cmd_args.end());
-  args = new_exec_args;
-  return frontline_parser->CvdArgs();
-}
-
-Result<std::unique_ptr<FrontlineParser>> FrontlineParser::Parse(
-    ParserParam param) {
-  CF_EXPECT(!param.all_args.empty());
-  std::unique_ptr<FrontlineParser> frontline_parser(new FrontlineParser(param));
-  CF_EXPECT(frontline_parser != nullptr,
-            "Memory allocation for FrontlineParser failed.");
-  CF_EXPECT(frontline_parser->Separate());
-  return frontline_parser;
-}
-
-FrontlineParser::FrontlineParser(const ParserParam& parser_param)
-    : server_supported_subcmds_{parser_param.server_supported_subcmds},
-      all_args_(parser_param.all_args) {}
+using selector::SeparateArguments;
+using selector::SeparatedArguments;
 
-Result<void> FrontlineParser::Separate() {
-  arguments_separator_ = CF_EXPECT(CallSeparator());
-  return {};
-}
+/* the very first command line parser
+ *
+ * Being aware of valid subcommands and cvd-specific commands, it will
+ * separate the command line arguments into:
+ *
+ *  1. program path/name
+ *  2. cvd-specific arguments
+ *     a) selector flags
+ *     b) non-selector flags
+ *  3. subcommand
+ *  4. subcommand arguments
+ */
```

Comment:
I think this applies to `SeparateArguments` instead.

Reply:
Fixed.